CodingTracker.AhmadJer99/CodingTracker/Controllers/OperationController.cs

```diff
@@ -0,0 +1,290 @@
+        AnsiConsole.Markup($"[green]You would need to code [lightgoldenrod2_1]{((float)WeeklyGoal / 7):0.0} hours[/] a day to achieve this goal![/]\n");
+
+        AnsiConsole.Markup($"[white](Press Any Key To Continue)[/]");
+        Console.ReadKey();
+    }
+    internal static void NewSession()
+    {
+        string startTime = "";
+        string endTime = "";
+        string sessionDate = "";
+
+        var userTimeEntryChoice = AnsiConsole.Prompt(
+            new SelectionPrompt<StartTimeOptions>()
+            .Title("[yellow]Do you want to start the session manually by entering start and end date or you can use a stopwatch[/]")
+            .AddChoices(Enum.GetValues<StartTimeOptions>()));
+
+        switch (userTimeEntryChoice)
+        {
+            case StartTimeOptions.StopWatch:
+                Console.WriteLine("Unsupported Yet");
+                StopWatch.StartClock();
+
+                startTime = Validation.ConvertTimeToString(StopWatch.ClockStartTime);
+                endTime = Validation.ConvertTimeToString(StopWatch.ClockEndTime);
+                break;
+
+            case StartTimeOptions.SpecificTime:
+                var start = Validation.AskValidTimeInput("Enter start time ");
+                var end = Validation.AskValidTimeInput("Enter end time ");
+                do
+                {
+                    if (end < start)
+                    {
+                        AnsiConsole.MarkupLine($"[red]Error: Invalid Input -End time cant be before starting time!!![/]");
+                        end = Validation.AskValidTimeInput("Enter end time ");
+                    }
+                }
+                while (end < start);
+
+                var date = Validation.AskValidDateInput("Enter session's date ");
+
+                sessionDate = Validation.ConvertDateToString(date);
+                startTime = Validation.ConvertTimeToString(start);
+                endTime = Validation.ConvertTimeToString(end);
+                break;
+        }
+
+        var newCodingSession = new CodingSession
+        {
+            SessionDate = sessionDate,
+            StartTime = startTime,
+            EndTime = endTime
+        };
+
+        DatabaseController.InsertRow(newCodingSession);
+
+    }
+
+    internal static void ViewAllSessions()
+    {
+
+        var codingSessions = DatabaseController.ReadAllRows(); // parse the rows into a list of coding sessions , each element is  a row in the table.
+
+        var columnNames = new List<string>() { "Id", "Session's Date", "Start Time", "End Time", "Duration(Hrs:Mins)" };
+        TableVisualisationEngine.ViewAsTable(codingSessions, TableAligntment.Center, columnNames);
+        // Pass the list of sessions into a table view engine , and show the results in a user friendly table.
+    }
+    internal static void FilteredView()
+    {
+        string filter = "";
```

Comment:
🟠 **Multiple String Concatenations**

💡 Contentious topic, but general rule of thumb is:
- When concatenating three dynamic string values or less, use traditional string concatenation.
- When concatenating more than three dynamic string values, use `StringBuilder`.

🔧 `var builder = new StringBuilder(); builder.Append("..."); filter = builder.ToString();`